contracts/liquidity_pool.cairo

```diff
@@ -251,45 +434,131 @@ func _mint_option_token_short{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, ra
-    // ie for call options only pool with ETH is used and for PUT options only the USDC is used
+    // Make sure the contract is the one that user wishes to trade
+    let (contract_option_type) = IOptionToken.option_type(option_token_address);
+    let (contract_strike) = IOptionToken.strike(option_token_address);
+    let (contract_maturity) = IOptionToken.maturity(option_token_address);
+    let (contract_option_side) = IOptionToken.side(option_token_address);
+
+    with_attr error_message("Required contract doesnt match the address.") {
+        assert contract_option_type = option_type;
+        assert contract_strike = strike;
+        assert contract_maturity = maturity;
+        assert contract_option_side = side;
+    }
 
-    // assuming we are in a CALL pool (ie ETH pool)
+    if (option_side == TRADE_SIDE_LONG) {
+        _burn_option_token_long(
+            currency_address=currency_address,
+            option_token_address=option_token_address,
+            amount=amount,
+            premia=premia,
+            fees=fees,
+        );
+    } else {
+        _burn_option_token_short(
+            currency_address=currency_address,
+            option_token_address=option_token_address,
+            amount=amount,
+            premia=premia,
+            fees=fees,
+            option_type=option_type,
+            underlying_price=underlying_price,
+        );
+    }
 
-    // if option_side == TRADE_SIDE_LONG:
-    //     _burn_option_token_long(address, amount, strike, maturity)
-    // else:
-    //     _burn_option_token_short(address, amount, strike, maturity)
-    // end
     return ();
 }
 
 func _burn_option_token_long{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-    address: felt, amount: felt, strike: felt, maturity: felt
+    currency_address: felt, option_token_address: felt, amount: felt, premia: felt, fees: felt
 ) {
-    // address is user address
-    // long_call_burn(address, amount, strike, maturity)
+    alloc_locals;
 
-    // user gets premia minus fees
-    // pool updates its available capital (unlocks it)
+    let current_contract_address = get_contract_address();
+    let user_address = get_caller_address();
+
+    // Burn the tokens
+    IOptionToken.burn(option_token_address, user_address, amount);
+
+    // Send (premia - fees)  to user
+    let to_be_received_by_user = premia - fees;
+
+    IERC20.transferFrom(
+        contract_address=currency_address,
+        sender=current_contract_address,
+        recipient=user_address,
+        amount=to_be_received_by_user,
+    );
+
+    // Increase available capital by (amount - (premia + fees))
+    let (current_unlocked_balance) = option_token_unlocked_capital.read(option_token_address);
+    let increase_unlocked_by = (amount - premia) + fees;
+    let new_unlocked_balance = current_unlocked_balance + increase_unlocked_by;
+    option_token_locked_capital.write(option_token_address, new_locked_balance);
```

Comment:
@Chepelau 
- few lines above there is `option_token_unlocked_capital.read` - UNLOCKED
- here we have `option_token_locked_capital` - LOCKED
- can you fix it? 

also... we have both of these storage_vars in this module... This was one of my stupid ideas that I didn't thing through (this option was)... this does account for supplying new capital for the pool... could you
- use only one of the storage_vars (ideally locked_capital - less prone to mess this up)
- get the unlocked_capital by getting diff from all given tokens in pool minus the locked_capital
```
IERC20.balanceOf(
        contract_address=pooled_token_addr, account=own_addr
    );
```